src/transformers/generation_utils.py

```diff
@@ -841,6 +841,80 @@ def compute_transition_beam_scores(
+                model_kwargs.pop(key, None)
+
+        # Transfo_XL does not use have "attention_mask" as an argument, and it is harmless (it is being passed in the
+        # tests, through)
+        if "transfoxl" in str(self).lower():
+            model_kwargs.pop("attention_mask", None)
+
+        unused_model_args = []
+        model_args = set(inspect.signature(self.prepare_inputs_for_generation).parameters)
+        # `kwargs`` if often used to handle optional forward pass inputs like `attention_mask`. If
+        # `prepare_inputs_for_generation` doesn't accept `kwargs`, then a stricter check can be made ;)
+        if "kwargs" in model_args:
+            model_args |= set(inspect.signature(self.forward).parameters)
+        for key, value in model_kwargs.items():
+            if value is not None and key not in model_args:
+                unused_model_args.append(key)
+
+        if unused_model_args:
+            raise ValueError(
+                f"The following `model_kwargs` are not used by the model: {unused_model_args} (note: typos in the"
+                " generate arguments will also show up in this list)"
+            )
+
+    def _validate_generation_inputs(
+        self,
+        generation_inputs: Dict[str, Any],
+        generation_method_name: str,
+        supporting_objects: List[Callable],
+    ):
+        """
+        Validates the generation inputs for the generation submethod. If there are arguments that were passed but will
+        not be used by the generation method or its supporting objects, an informative exception will be thrown. A
+        supporting object is a class or function that precedes the generation submethod itself.
+        """
+        # Excludes arguments that are handled before calling the submethods
+        unused_keys = ["self"]
+        keys_processed_before_method = [
+            "inputs",
+            "bos_token_id",
+            "max_new_tokens",
+            "decoder_start_token_id",
+            "do_sample",
+            "model_kwargs",
+        ]
+        keys_passed_to_model_kwargs = ["use_cache", "output_attentions", "output_hidden_states"]
+        for key in unused_keys + keys_processed_before_method + keys_passed_to_model_kwargs:
+            generation_inputs.pop(key)
+
+        # Validates that the remainder arguments are used by the generation submethod
+        unused_args = []
+        generation_method = getattr(self, generation_method_name)
+        generation_method_args = set(inspect.signature(generation_method).parameters)
+        for supporting_object in supporting_objects:
+            generation_method_args |= set(inspect.signature(supporting_object).parameters)
+        # Ad hoc replacement of supporting object argument to match the corresponding generation argument
+        if "do_early_stopping" in generation_method_args:
+            generation_method_args.remove("do_early_stopping")
+            generation_method_args.add("early_stopping")
```

Comment:
The `early_stopping` is consumed by a class that has `do_early_stopping` as argument. Since the class is public, I can't touch it :( At most I can add a new argument doing the same, but probably not worth it.